File: src/vpc/internal/subscriptionsUpsert.js
"use strict";

const { verifyInternalSignature } = require("../../common/auth");

async function handleInternalSubscriptionsUpsert({ event, method, res }) {
  if (method !== "POST") return res.methodNotAllowed("Method not allowed");

  const v = verifyInternalSignature(event);
  if (!v.ok) return res.forbidden(`Internal auth failed: ${v.reason}`);

  return res.notImplemented("TODO: internal subscriptions upsert (write RDS)");
}

module.exports = { handleInternalSubscriptionsUpsert };

File: src_C/Vpc/Internal/SubscriptionsUpsert.cs
using Amazon.Lambda.APIGatewayEvents;
using RecallSmith.Lambda.Common;

namespace RecallSmith.Lambda.Vpc.Internal;

public static class SubscriptionsUpsert
{
  public static Task<APIGatewayProxyResponse> HandleInternalSubscriptionsUpsert(LambdaRequest req, Res res)
  {
    if (req.Method != "POST") return Task.FromResult(res.MethodNotAllowed("Method not allowed"));

    var v = Auth.VerifyInternalSignature(req);
    if (!v.Ok) return Task.FromResult(res.Forbidden($"Internal auth failed: {v.Reason}"));

    return Task.FromResult(res.NotImplemented("TODO: internal subscriptions upsert (write RDS)"));
  }
}

